BrainLILODrv.cpp

```diff
@@ -173,12 +173,82 @@ static void ShowMessage(std::wstring msg, std::wstring title, UINT typ)
     MessageBox(NULL, msg.c_str(), title.c_str(), typ);
 }
 
+static void SetAddress(Brain_Gen gen)
+{
+    switch (gen)
+    {
+    case Gen1:
+        // VA
+        bootloader_preload_address = BOOTLOADER_PRELOADADDR_GEN1;
+        bootloader_load_address = BOOTLOADER_LOADADDR_GEN1;
+        physical_invoker_install_address = PHYSICAL_INVOKER_INSTALLADDR_GEN1;
+        // PA
+        bootloader_load_address_pa = BOOTLOADER_LOADADDR_GEN1_PA;
+        break;
+    default:
+        // Gen2?
+        // VA
+        bootloader_preload_address = BOOTLOADER_PRELOADADDR;
+        bootloader_load_address = BOOTLOADER_LOADADDR;
+        physical_invoker_install_address = PHYSICAL_INVOKER_INSTALLADDR;
+        // PA
+        bootloader_load_address_pa = BOOTLOADER_LOADADDR_PA;
+    }
+}
+
+static Brain_Gen SelectGen(std::wstring model)
+{
+    Brain_Gen brain_gen = UnknownGen;
+    if (model == L"gen1.bin")
+    {
+        brain_gen = Gen1;
+    }
+    else if (model == L"gen2.bin")
+    {
+        brain_gen = Gen2;
+    }
+    else if (model == L"gen3_1.bin")
+    {
+        brain_gen = Gen3_1;
+    }
+    else if (model == L"gen3_2.bin")
+    {
+        brain_gen = Gen3_2;
+    }
+    else if (model == L"gen3_3.bin")
+    {
+        brain_gen = Gen3_3;
+    }
+    else if (model == L"gen3_4.bin")
+    {
+        brain_gen = Gen3_4;
+    }
+    else if (model == L"gen3_5.bin")
+    {
+        brain_gen = Gen3_5;
+    }
+    else if (model == L"gen3_6.bin")
+    {
+        brain_gen = Gen3_6;
+    }
+    else if (model == L"gen3_7.bin")
+    {
+        brain_gen = Gen3_7;
+    }
+    else
+    {
+        brain_gen = UnknownGen;
+    }
+    return brain_gen;
+}
+
 static bool doLinux()
 {
     std::wifstream iVersion;
     std::wstring line, model;
     std::wregex modelRe(L"[A-Z]{2}-[A-Z0-9]+");
     std::wsmatch match;
+    Brain_Gen brain_gen = UnknownGen;
```

Comment:
(must) Please use lowerCamelCase to name variables.
```suggestion
    BrainGen brainGen = UnknownGen;
```